tests/common/fixtures/elasticsearch.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import os
+
+import pytest
+
+from h import search
+
+ELASTICSEARCH_HOST = os.environ.get("ELASTICSEARCH_HOST", "http://localhost:9200")
+ELASTICSEARCH_INDEX = "hypothesis-test"
+
+
+@pytest.fixture
+def es_client():
+    """A :py:class:`h.search.client.Client` for the test search index."""
+    return _es_client()
+
+
+@pytest.fixture(scope="session", autouse=True)
+def init_elasticsearch(request):
+    """Initialize the test Elasticsearch index once per test session."""
+    client = _es_client()
+
+    def maybe_delete_index():
+        """Delete the test index if it exists."""
+        if client.conn.indices.exists(index=ELASTICSEARCH_INDEX):
+            client.conn.indices.delete(index=ELASTICSEARCH_INDEX)
+
+    # Delete the test search index at the end of the test run.
+    request.addfinalizer(maybe_delete_index)
+
+    # Delete the test search index at the start of the run, just in case it
+    # was somehow left behind by a previous test run.
+    maybe_delete_index()
+
+    # Initialize the test search index.
+    search.init(client)
+
+
+@pytest.fixture(autouse=True)
```

Comment:
Making this fixture run after every single test, including those which do not use elasticsearch at all, adds overhead to the test suite. I suggest we find a way to only run this after tests which use ES or alternatively, before tests which use ES.

On my system, the following reports a total of ~20 seconds for the whole `h` test suite, which doesn't include the time spent creating the client. Overall the unit tests take ~123s on this branch vs ~100 on master. Run with `tox -- --capture=no tests/h/`:

```diff
diff --git a/tests/common/fixtures/__init__.py b/tests/common/fixtures/__init__.py
index 530984d66..522e357a5 100644
--- a/tests/common/fixtures/__init__.py
+++ b/tests/common/fixtures/__init__.py
@@ -4,10 +4,12 @@ from __future__ import unicode_literals
 from tests.common.fixtures.elasticsearch import es_client
 from tests.common.fixtures.elasticsearch import init_elasticsearch
 from tests.common.fixtures.elasticsearch import delete_all_elasticsearch_documents
+from tests.common.fixtures.elasticsearch import log_total_delete_time
 
 
 __all__ = (
     "es_client",
     "init_elasticsearch",
     "delete_all_elasticsearch_documents",
+    "log_total_delete_time",
 )
diff --git a/tests/common/fixtures/elasticsearch.py b/tests/common/fixtures/elasticsearch.py
index 0911552db..081feb164 100644
--- a/tests/common/fixtures/elasticsearch.py
+++ b/tests/common/fixtures/elasticsearch.py
@@ -1,6 +1,7 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
 
+import time
 import os
 
 import pytest
@@ -38,17 +39,30 @@ def init_elasticsearch(request):
     search.init(client)
 
 
+total_delete_time = 0
+
+
 @pytest.fixture(autouse=True)
 def delete_all_elasticsearch_documents(request):
     """Delete everything from the test search index after each test."""
     client = _es_client()
 
     def delete_everything():
+        global total_delete_time
+        start = time.time()
         client.conn.delete_by_query(index=client.index, body={"query": {"match_all": {}}})
+        end = time.time()
+        total_delete_time += (end-start)
 
     request.addfinalizer(delete_everything)
 
 
+@pytest.fixture(scope='session', autouse=True)
+def log_total_delete_time():
+    yield
+    print('total delete time {}'.format(total_delete_time))
+
+
 def _es_client():
     """Return a :py:class:`h.search.client.Client` for the test search index."""
     return search.get_client({"es.host": ELASTICSEARCH_HOST, "es.index": ELASTICSEARCH_INDEX})
diff --git a/tests/h/conftest.py b/tests/h/conftest.py
index bd3cf3da9..bbeb2a0ba 100644
--- a/tests/h/conftest.py
+++ b/tests/h/conftest.py
@@ -27,6 +27,7 @@ from h._compat import text_type
 from tests.common.fixtures import es_client  # noqa: F401
 from tests.common.fixtures import init_elasticsearch  # noqa: F401
 from tests.common.fixtures import delete_all_elasticsearch_documents  # noqa: F401
+from tests.common.fixtures import log_total_delete_time  # noqa: F401
 
 TEST_AUTHORITY = 'example.com'
 TEST_DATABASE_URL = database_url(os.environ.get('TEST_DATABASE_URL',
```